### ISSUE TITLE
Unreachable State Not Caught

### ISSUE THREAD

**Issue by mtsvickers** [airware/asl-validator#17]
If a state us unreachable it should be considered invalid ASL, but right now still validates.
Ex: ( "Finished Choice" is unreachable but this returns valid)
```
{
	"StartAt": "Start State",
	"States":{
		"Start State": {
			"Type": "Pass",
			"Next": "Finished"
		},
		"Finished Choice":{
			"Type":"Choice",
			"Choices": [
				{
					"Variable": "$.x",
					"BooleanEquals": false,
					"Next": "Finished"
				}
			],
			"Default": "Finished"
		},
		"Finished": {
			"Type": "Pass",
			"End": true
		}
	}
}
```

**Comment by ChristopheBougere** [comment]
Thanks for reporting!

![image](https://user-images.githubusercontent.com/1325474/37514084-3d60ceb0-2907-11e8-9a7a-edd12e3eb82b.png)

You're right, this state machine definition should throw
```
Error code: MISSING_TRANSITION_TARGET 
Message: State "Finished Choice" is not reachable.
```

For now, `asl-validator` is validating syntax statically using JSON schemas and JSON paths.
I'd like to add this case, but we need to define a clear rule to know when a state is unreachable.
Maybe every state should be referenced either in:
- a `StartAt` field (of the state machine or a parallel state)
- a `Next` field (of a state or a catch)
- a `Default` field (of a choice)

Do you see anything else?

**Comment by mtsvickers** [comment]
I think that would cover everything

**Comment by mtsvickers** [comment]
Thank you!

On Wed, Apr 4, 2018 at 4:21 AM, Christophe Bougère <notifications@github.com
